lib/resty/rediscluster.lua

```diff
@@ -171,130 +374,65 @@ local function try_hosts_slots(self, serv_list)
+                table_insert(errors, status_err)
+                return nil, errors
+            end
 
-                    for slot = start_slot, end_slot do
-                        -- todo: there is no need to create the 'list', as all the slots
-                        -- within [start_slot, end_slot] share the same 'servers'.
-                        -- just do: slots[slot] = servers
-                        local list = { serv_list = {} }
-                        -- from 3, here lists the host/port/nodeid of in charge nodes
-                        for j = 3, #sub_info do
-                            sub_info_j = sub_info[j]
-                            sub_info_j_ip = sub_info_j[4]
-                            if sub_info_j_ip and type(sub_info_j_ip) == "table" and
-                               next(sub_info_j_ip) and sub_info_j_ip[2] ~= sub_info_j[1]
-                            then
-                                -- prioritize IP address over hostname as IP does not suffer from stale DNS cache
-                                list.serv_list[#list.serv_list + 1] = { ip = sub_info_j_ip[2], port = sub_info_j[2] }
-                            end
-                            list.serv_list[#list.serv_list + 1] = { ip = sub_info_j[1], port = sub_info_j[2] }
-
-                            slots[slot] = list
-                        end
-                    end
-                end
-                ngx_log(ngx_debug, "finished initializing slotcache ...")
-                slot_cache[self.config.name] = slots
-                slot_cache[self.config.name .. "serv_list"] = servers
-            else
-                ngx_log(ngx_debug, "failed to initialize slotcache")
+            local _, version_err = check_version(self, redis_client)
+            if version_err then
                 redis_client:close()
-                table_insert(errors, err)
+                table_insert(errors, version_err)
                 return nil, errors
             end
 
-            -- cache master nodes
-            local nodes_res, nerr = redis_client:cluster("nodes")
-            if nodes_res then
-                local nodes_info = split(nodes_res, char(10))
-                for _, node in ipairs(nodes_info) do
-                    local node_info = split(node, " ")
-                    if #node_info > 2 then
-                        local is_master = match(node_info[3], "master") ~= nil
-                        if is_master then
-                            local ip_port = split(split(node_info[2], "@")[1], ":")
-                            table_insert(master_nodes, {
-                                ip = ip_port[1],
-                                port = tonumber(ip_port[2])
-                            })
-                        end
-                    end
-                end
-            else
-                ngx_log(ngx_debug, "failed to fetch master nodes")
+            --[=[
+            -- deprecated this by `parse_slots()`
```

Comment:
Unless we know for certain what are the upcoming changes in Redis 8.x version and that this function will be needed I'd say we don't need this comment and we can remove the entire `parse_masters` function as well. Following YAGNI rule.

Reply:
Let's keep it.

Reply:
done